src/Foreign/Generic/Class.purs

```diff
@@ -95,10 +100,16 @@ instance genericEncodeConstructor
         else case opts.sumEncoding of
                TaggedObject { tagFieldName, contentsFieldName, constructorTagTransform } ->
                  unsafeToForeign (Object.singleton tagFieldName (unsafeToForeign $ constructorTagTransform ctorName)
-                           `Object.union` maybe Object.empty (Object.singleton contentsFieldName) (encodeArgsArray args))
+                           `Object.union` objectFromArgs opts.sumEncoding (encodeArgsArray args))
     where
       ctorName = reflectSymbol (SProxy :: SProxy name)
 
+      objectFromArgs :: SumEncoding -> Maybe Foreign -> Object Foreign
+      objectFromArgs _ Nothing = Object.empty
+      objectFromArgs (TaggedObject { contentsFieldName, unwrapRecords }) (Just f)
+        | typeOf f == "object" && not isArray f && unwrapRecords = unsafeFromForeign f
```

Comment:
I feel like it'd be better to type-case here, instead of matching on the runtime representation of the JSON. That way we know we only unwrap actual PureScript records, which arguably makes things slightly more predictable. 

Does Aeson do a similar thing here?

Reply:
Anything I can do to help shepherd this along to be merged?

Reply:
sorry for the lack of input but we will shortly be updating this a little bit I think to make sure it can be merged

Reply:
@paf31 I'm back on this, I've added List and some small things in another branch but I want to get this merged first. I don't really understand what you mean?